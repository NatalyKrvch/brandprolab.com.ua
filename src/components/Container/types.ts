export interface ContainerProps {
  fullWidth?: boolean;
}
