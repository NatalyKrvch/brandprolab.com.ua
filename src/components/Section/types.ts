import type { HTMLAttributes } from 'react';

import { ContainerVariant } from '../Container';

export interface SectionProps extends HTMLAttributes<HTMLElement> {
  containerClassName?: string;
  containerVariant?: ContainerVariant;
}
