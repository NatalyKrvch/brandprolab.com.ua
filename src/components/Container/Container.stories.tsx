import type { Meta, StoryObj } from '@storybook/react';
import Container from './Container';

const meta: Meta<typeof Container> = {
  title: 'Components/Container',
  component: Container,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    fullWidth: {
      control: { type: 'boolean' },
      description:
        'If true, the container will span the full width of the screen.',
    },
    children: {
      control: { type: 'text' },
      description: 'The content to be rendered inside the container.',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Container>;

export const Default: Story = {
  args: {
    fullWidth: false,
    children: (
      <div className="text-base">
        This is a default container with max-width 1224px.
      </div>
    ),
  },
};

export const FullWidth: Story = {
  args: {
    fullWidth: true,
    children: <div className="text-base">This is a full-width container.</div>,
  },
};
