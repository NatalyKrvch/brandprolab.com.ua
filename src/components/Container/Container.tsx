import {
  FULL_WIDTH_CONTAINER_TEST_ID,
  LIMITED_WIDTH_CONTAINER_TEST_ID,
} from '@/lib/constants';
import { ContainerProps } from './types';

const Container = ({ fullWidth, children }: ContainerProps) => {
  if (fullWidth) {
    return (
      <div className="mx-auto" data-testid={FULL_WIDTH_CONTAINER_TEST_ID}>
        {children}
      </div>
    );
  }

  return (
    <div
      className="mx-auto max-w-[1224px] px-4 tablet:px-12 desktop:px-12"
      data-testid={LIMITED_WIDTH_CONTAINER_TEST_ID}
    >
      {children}
    </div>
  );
};

export default Container;
