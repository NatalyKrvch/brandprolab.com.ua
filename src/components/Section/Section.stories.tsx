import type { Meta, StoryObj } from '@storybook/react';

import { ContainerVariant } from '@/components/Container';

import Section from './Section';

const meta: Meta<typeof Section> = {
  title: 'Layout/Section',
  component: Section,
  tags: ['autodocs'],
  argTypes: {
    containerVariant: {
      control: { type: 'select' },
      options: Object.values(ContainerVariant),
    },
    className: {
      control: 'text',
    },
    containerClassName: {
      control: 'text',
    },
    children: {
      control: 'text',
    },
  },
};

export default meta;

type Story = StoryObj<typeof Section>;

export const Default: Story = {
  args: {
    children: <p>Default section content</p>,
  },
};

export const FullWidthContainer: Story = {
  args: {
    containerVariant: ContainerVariant.FULL,
    children: <p>Full width content</p>,
  },
};

export const WithCustomClasses: Story = {
  args: {
    className: 'bg-gray-100 py-8',
    containerClassName: 'border border-dashed border-red-500',
    children: <p>Section with custom styles</p>,
  },
};
