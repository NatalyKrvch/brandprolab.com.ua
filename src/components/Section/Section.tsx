import type { PropsWithChildren } from 'react';

import { Container } from '@/components';
import { ContainerVariant } from '@/components/Container';

import type { SectionProps } from './types';

const Section = ({
  children,
  className,
  containerClassName,
  containerVariant = ContainerVariant.CENTERED,
  ...props
}: PropsWithChildren<SectionProps>) => {
  return (
    <section className={className} {...props}>
      <Container variant={containerVariant} className={containerClassName}>
        {children}
      </Container>
    </section>
  );
};

export default Section;
