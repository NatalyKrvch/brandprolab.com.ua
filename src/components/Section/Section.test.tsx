import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import {
  FULL_WIDTH_CONTAINER_TEST_ID,
  LIMITED_WIDTH_CONTAINER_TEST_ID,
} from '@/lib/testIDs';

import { ContainerVariant } from '../Container';
import Section from './Section';

jest.mock('@/sanity/lib/serverClient', () => ({
  client: { fetch: jest.fn() },
}));

jest.mock('@/sanity/lib/image', () => ({
  urlForImage: (image: any) =>
    image?.asset?._ref ? `/mocked-url-${image.asset._ref}` : '/mocked-fallback',
}));

jest.mock('yet-another-react-lightbox', () => {
  const MockLightbox = () => (
    <div data-testid="mock-lightbox">Mock Lightbox</div>
  );
  MockLightbox.displayName = 'MockLightbox';
  return MockLightbox;
});

jest.mock('yet-another-react-lightbox/plugins/zoom', () => ({}));

describe('Section component', () => {
  it('renders children inside the section and default container', () => {
    const { asFragment } = render(
      <Section>
        <p>Section content</p>
      </Section>,
    );

    const container = screen.getByTestId(LIMITED_WIDTH_CONTAINER_TEST_ID);
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent('Section content');

    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className and containerClassName', () => {
    const { container, asFragment } = render(
      <Section className="bg-red-500" containerClassName="py-8">
        <p>Styled section</p>
      </Section>,
    );

    const section = container.querySelector('section');
    expect(section).toHaveClass('bg-red-500');

    const containerElement = screen.getByTestId(
      LIMITED_WIDTH_CONTAINER_TEST_ID,
    );
    expect(containerElement).toHaveClass('py-8');

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with full-width container when variant is FULL', () => {
    const { asFragment } = render(
      <Section containerVariant={ContainerVariant.FULL}>
        <p>Full width section</p>
      </Section>,
    );

    const container = screen.getByTestId(FULL_WIDTH_CONTAINER_TEST_ID);
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent('Full width section');

    expect(asFragment()).toMatchSnapshot();
  });
});
