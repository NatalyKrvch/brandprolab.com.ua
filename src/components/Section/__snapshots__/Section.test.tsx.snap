// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component applies custom className and containerClassName 1`] = `
<DocumentFragment>
  <section
    class="bg-red-500"
  >
    <div
      class="mx-auto max-w-container px-4 tablet:px-12 desktop:px-12 py-8"
      data-testid="limited-width-container-test-id"
    >
      <p>
        Styled section
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section component renders children inside the section and default container 1`] = `
<DocumentFragment>
  <section>
    <div
      class="mx-auto max-w-container px-4 tablet:px-12 desktop:px-12"
      data-testid="limited-width-container-test-id"
    >
      <p>
        Section content
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Section component renders with full-width container when variant is FULL 1`] = `
<DocumentFragment>
  <section>
    <div
      class="w-full"
      data-testid="full-width-container-test-id"
    >
      <p>
        Full width section
      </p>
    </div>
  </section>
</DocumentFragment>
`;
